### DISCUSSION TITLE
why sign client not redirect me to the dapp  when i trigger the transaction  through wallet

### DISCUSSION THREAD

**Discussion by arbaz-meo** [WalletConnect/walletconnect-monorepo#3451] [Developer Support] [answered]
i have one doubt or maybe i m missing something 

when i connect wallet by signclient its open the wallet automatically by using deep link **### https://metamask.app.link/wc?uri=encode wcuri**   .
 i have wc uri so i genrate deep link on connection but when i go to trigger the transaction i have this deep link https://metamask.app.link  so its take me to the wallet but when i trigger transaction its not bring me back on dapp. 

is i am missing something ???

**Comment by arbaz-meo** [comment]
@glitch-txs can you give me idea of this

**Reply by glitch-txs** [reply, marked as answer]
Hi this is an issue on MetaMask side, you can open an issue on their repo :)

**Reply by arbaz-meo** [reply]
@glitch-txs this is known issue ???

**Comment by mshahjahan-mufin** [comment]
@arbaz-meo Instead of deep link use app link which is metamask://. Sometimes, MetaMask behaves like this. If you are not using any wallet other than MetaMask then directly go for their DeepLink. You can use the parameters from here: [Meta Mask Depp link](https://metamask.github.io/metamask-deeplinks/#)
